File: types/datasets.go
package types

type NewClass struct{
	Name				string	`json:"name"`
	ScientificName		string	`json:"scientific_name"`
	Description			string	`json:"description"`
	Status				string	`json:"status"`
}


type Fetch_DatasetClass struct{
	ID 					int64 	`json:"class_id"`
	Name				string	`json:"name"`
	ScientificName		string	`json:"scientific_name"`
	Description			string	`json:"description"`
	Status				string	`json:"status"`
	Count				int		`json:"count"`
}

type EditClass struct{
	ID 					int64 	`json:"class_id"`
	ScientificName		string	`json:"scientific_name"`
	Description			string	`json:"description"`
	Status				string	`json:"status"`

}

type Train struct{
	Version		string `json:"version"`
}

File: client/src/types/datasets.ts
export type DatasetClassTypes = {
    class_id:           number
    name:               string 
    scientific_name:    string
    description:        string 
    status:             string 
    count:              string 
}

export type DeleteDatasetClassType = {
    class_id: number, 
    className:string
}


export type FetchModelType = {
    model_id: number
    version: string,
    train_acc: number,
    train_loss: number,
    trained_at: string
}


export type ImageData = {
    key: string,
    url: string
}


export type DeleteImageDataTypes = { 
    selectedKeys: string[], 
    class_id: number, 
    datasetClass: string 
}